src/position.h

```diff
@@ -268,6 +269,10 @@ inline Square Position::ep_square() const {
   return st->epSquare;
 }
 
+inline Square Position::king_square(Color c) const {
+  return lsb(pieces(c, KING));
```

Comment:
Please add a kingSquare variable in StateInfo and use that. It should be much faster both in 64 and 32 bit cases.